src/zmq/zmqnotificationinterface.cpp

```diff
@@ -75,14 +75,18 @@ bool CZMQNotificationInterface::Initialize()
     LogPrint(BCLog::ZMQ, "zmq: Initialize notification interface\n");
     assert(!pcontext);
 
-    pcontext = zmq_init(1);
+    pcontext = zmq_ctx_new();
 
     if (!pcontext)
     {
         zmqError("Unable to initialize context");
         return false;
     }
 
+    int major = 0, minor = 0, patch = 0;
+    zmq_version(&major, &minor, &patch);
```

Comment:
Can be called before context creation?

Reply:
Yes, in practice, it seems that it can be. The reason why I still put it after context creation is because of this quote in the API docs at http://api.zeromq.org/master:zmq: `Before using any ØMQ library functions you must create a ØMQ context.`

Reply:
It can matter if `zmq_ctx_new` fails for some reason, then you don't get the version in the logs.

Reply:
Makes sense. Updated, thanks!